codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostRequestDecoderTest.java

```diff
@@ -976,4 +978,244 @@ public void testDecodeMultipartRequest() {
             assertTrue(req.release());
         }
     }
+
+    private HttpPostMultipartRequestDecoder commonTestBigFileDelimiterInMiddleChunk(HttpDataFactory factory,
+                                                                                    HttpRequest request, String prefix,
+                                                                                    String suffix1, String suffix2,
+                                                                                    int nbChunks, int bytesPerChunk,
+                                                                                    int bytesLastChunk) {
+        HttpPostMultipartRequestDecoder decoder = new HttpPostMultipartRequestDecoder(factory, request);
+        decoder.offer(new DefaultHttpContent(Unpooled.wrappedBuffer(prefix.getBytes(CharsetUtil.UTF_8))));
+
+        byte[] body = new byte[bytesPerChunk];
+        Arrays.fill(body, (byte) 1);
+        for (int i = 0; i < nbChunks; i++) {
+            ByteBuf content = Unpooled.wrappedBuffer(body, 0, bytesPerChunk);
+            decoder.offer(new DefaultHttpContent(content)); // **OutOfMemory here**
+            content.release();
+        }
+
+        byte[] bsuffix1 = suffix1.getBytes(CharsetUtil.UTF_8);
+        byte[] lastbody = new byte[bytesLastChunk + bsuffix1.length];
+        Arrays.fill(body, (byte) 1);
+        for (int i = 0; i < bsuffix1.length; i++) {
+            lastbody[bytesLastChunk + i] = bsuffix1[i];
+        }
+
+        ByteBuf content2 = Unpooled.wrappedBuffer(lastbody, 0, lastbody.length);
+        decoder.offer(new DefaultHttpContent(content2));
+        content2.release();
+        content2 = Unpooled.wrappedBuffer(suffix2.getBytes(CharsetUtil.UTF_8));
+        decoder.offer(new DefaultHttpContent(content2));
+        content2.release();
+        decoder.offer(new DefaultLastHttpContent());
+        return decoder;
+    }
+
+    @Test
+    public void testBIgFileUploadDelimiterInMiddleChunkDecoder() throws IOException {
+        int nbChunks = 100;
+        int bytesPerChunk = 1000000;
+        int bytesLastChunk = 10000;
+        int fileSize = bytesPerChunk * nbChunks + bytesLastChunk; // set Xmx to a number lower than this and it crashes
```

Comment:
The test data is 100 MB. Can't we get meaningful coverage with something smaller?

Reply:
It was the original code provided by issue submitter.
But I agree, we can limit the size for something like 10 MB (using 100KB chunk size)

Reply:
Done and all tests passed